lib/model/content.dart

```diff
@@ -808,10 +808,12 @@ class _ZulipContentParser {
   }
 
   static final _userMentionClassNameRegexp = () {
-    // This matches a class `user-mention` or `user-group-mention`,
-    // plus an optional class `silent`, appearing in either order.
-    const mentionClass = r"user(?:-group)?-mention";
-    return RegExp("^(?:$mentionClass(?: silent)?|silent $mentionClass)\$");
+    // This matches the classes `user-mention`, `user-group-mention`,
+    // or `user-mention channel-wildcard-mention`, plus an optional
+    // class `silent`, appearing in either order.
+    const mentionClass = r"user(?:-group)?-mention|"
+                          "(?:user-mention channel-wildcard-mention)";
+    return RegExp("^(?:$mentionClass)(?: silent)?|silent (?:$mentionClass)\$");
```

Comment:
Looks like the non-capturing groups added here are not necessary:
```suggestion
    const mentionClass = r"user(?:-group)?-mention|"
                          "user-mention channel-wildcard-mention";
    return RegExp("^(?:$mentionClass)(?: silent)?|silent $mentionClass\$");
```

Reply:
Seeing this I wondered if regular expressions are the right tool for this issue, but then I read from the commit message of 4a10b3cf8cb286396e034a830993c32544ed0726 mentioning that this is more efficient than `classes`. Makes sense.

Reply:
The groups around `mentionClass` are needed. The revised version above would have a three-way alternative: either `^(?:$mentionClass)(?: silent)?`, or `silent user(?:-group)?-mention`, or `user-mention channel-wildcard-mention$`.

Probably the cleanest way to express that is by `mentionClass` itself containing the `(?:` and `)`.

In main there's also a group enclosing everything between `^` and `$`. That's necessary so that `^` and `$` apply to all alternatives, instead of just the first and last of them respectively.

Reply:
I've just sent #1086 to refactor the existing logic here, reducing the role of the regexp and hopefully making it easier to add more features. @sm-sayedi PTAL and try rebasing your work atop that.